airflow/dag_processing/processor.py

```diff
@@ -484,7 +486,13 @@ def manage_slas(self, dag: DAG, session: Session = None) -> None:
                         callback(dag, task_list, blocking_task_list, slas, blocking_tis)
                         notification_sent = True
                     except Exception:
-                        Stats.incr("sla_callback_notification_failure")
+                        Stats.incr(
+                            "sla_callback_notification_failure",
+                            tags={
+                                "dag_id": dag.dag_id,
+                                "func_name": callback.func_name,  # type: ignore[attr-defined]
```

Comment:
This is not really related, but can anyone tell me where this `func_name` attribute is set? I searched the entire code base and there’s not a trace of it. The only other place that seems to reference this attribute (aside from this block) is a test `test_dag_file_processor_sla_miss_callback_exception`, and in that test the attribute does not actually exist and is only passing because the “callback function” is a mock there.

https://github.com/apache/airflow/blob/d5ac1b057f96d94062a92d968fe40f4371eb5da0/tests/dag_processing/test_processor.py#L328-L332

Reply:
Actually this is the way to get the callback name in python 2, but it's removed in python 3 and replaced by `callback.__name__`, so we need to fix this.
Would you prefer I fix it here or in a separate PR?

Reply:
Ah that makes sense. I think you’ll need to fix that here since otherwise your code will crash the DAG processor (previously the processor would only crash if it reaches the `except` block, which is much rarer).